pkg/apiclient/offlineclient.go

```diff
@@ -0,0 +1,49 @@
+package apiclient
+
+import (
+	"context"
+
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient/clusterworkflowtemplate"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient/cronworkflow"
+	infopkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/info"
+	workflowpkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflow"
+	workflowarchivepkg "github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflowarchive"
+	"github.com/argoproj/argo-workflows/v3/pkg/apiclient/workflowtemplate"
+)
+
+type offlineClient struct {
+}
+
+var _ Client = &offlineClient{}
+
+func newOfflineClient() (context.Context, Client, error) {
+	return context.Background(), &offlineClient{}, nil
+}
+
+func (a *offlineClient) NewWorkflowServiceClient() workflowpkg.WorkflowServiceClient {
+	return &errorTranslatingWorkflowServiceClient{nil}
+}
+
+func (a *offlineClient) NewOfflineWorkflowServiceClient() OfflineWorkflowServiceClient {
+	return OfflineWorkflowServiceClient{}
+}
+
+func (a *offlineClient) NewCronWorkflowServiceClient() cronworkflow.CronWorkflowServiceClient {
+	return &errorTranslatingCronWorkflowServiceClient{nil}
```

Comment:
ah - this will panic obvs - maybe these funcs now need to return `(CronWorkflowServiceClient, error)`?

Reply:
yes, that make sense now. Will update.